utils/hooks.ts

```diff
@@ -0,0 +1,18 @@
+import { useState, useEffect } from "react";
+
+export const useMediaQuery = (query: string) => {
+  const [matches, setMatches] = useState(false);
+
+  useEffect(() => {
+    const media = window.matchMedia(query);
```

Comment:
I think this should run as a layout effect, as it will effectively change the look of the page.
Since you can't run `useLayoutEffect` on the server, you can do something like this:
```
import { useEffect, useLayoutEffect } from 'react';

const useIsomorphicLayoutEffect =
  typeof window !== 'undefined' ? useLayoutEffect : useEffect;

// run the effect now
```

Reply:
I like that 👍  thx